sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -288,78 +290,85 @@ public synchronized void close() {
         return null;
       }
 
-      SplitReadStreamRequest splitRequest =
-          SplitReadStreamRequest.newBuilder()
-              .setName(source.readStream.getName())
-              .setFraction((float) fraction)
-              .build();
-
-      SplitReadStreamResponse splitResponse = storageClient.splitReadStream(splitRequest);
-      if (!splitResponse.hasPrimaryStream() || !splitResponse.hasRemainderStream()) {
-        // No more splits are possible!
-        Metrics.counter(
-                BigQueryStorageStreamReader.class,
-                "split-at-fraction-calls-failed-due-to-impossible-split-point")
-            .inc();
-        LOG.info(
-            "BigQuery Storage API stream {} cannot be split at {}.",
-            source.readStream.getName(),
-            fraction);
-        return null;
-      }
+      if (splitPossible) {
```

Comment:
Nit: I would structure this check as an early return (e.g. ```if (!splitPossible) { return null; }```) rather than putting the whole implementation into a new block.

Reply:
Done.